lib/worker.py

```diff
@@ -353,11 +358,17 @@ def run_loop(self, task_queue, result_queue):
                 retries_left = retries_left - 1
 
             result_queue.put(self.wrap_result(task_id, short_status, result_checksum))
-            if not lib.Options().args.is_force and short_status == 'fail':
-                color_stdout(
-                    'Worker "%s" got failed test; stopping the server...\n'
-                    % self.name, schema='test_var')
-                raise VoluntaryStopException()
+            if short_status == 'fail':
+                if lib.Options().args.is_force:
+                    self.restart_server(task_queue)
+                    color_stdout(
```

Comment:
Prints in both cases, just move it up (before if...else).

Reply:
"restart" message is the new one and only for "force" tests mode. Message "stop" was never printed before for force mode, also with "restart" message it's not needed.